src/core/skills/characters/yijiang2011/xuanhuo.ts

```diff
@@ -1,192 +1,187 @@
-        toId: fromId,
-        players: targetIds,
-        requiredAmount: 1,
-        conversation: TranslationPack.translationJsonPatcher(
-          'xuanhuo: please choose a target who {0} can slash',
-          TranslationPack.patchPlayerInTranslation(to),
-        ).extract(),
-      };
+  public async onTrigger() {
+    return true;
+  }
 
-      room.notify(
-        GameEventIdentifiers.AskForChoosingPlayerEvent,
-        EventPacker.createUncancellableEvent<GameEventIdentifiers.AskForChoosingPlayerEvent>(askForChoosingPlayer),
-        fromId,
-      );
+  public async onEffect(room: Room, event: ServerEventFinder<GameEventIdentifiers.SkillEffectEvent>) {
+    const { fromId, toIds, cardIds } = event;
+    const first = toIds![0];
+    const second = toIds![1];
 
-      const { selectedPlayers } = await room.onReceivingAsyncResponseFrom(
-        GameEventIdentifiers.AskForChoosingPlayerEvent,
-        fromId,
-      );
+    await room.moveCards({
+      movingCards: cardIds!.map(card => ({ card, fromArea: CardMoveArea.HandArea })),
+      fromId,
+      toId: first,
+      toArea: CardMoveArea.HandArea,
+      moveReason: CardMoveReason.ActivePrey,
+      proposer: fromId,
+      movedByReason: this.Name,
+    });
 
-      const victimId = selectedPlayers![0];
-      const askForUseSlashTo: ServerEventFinder<GameEventIdentifiers.AskForCardUseEvent> = {
-        toId,
-        cardUserId: toId,
-        scopedTargets: [victimId],
-        cardMatcher: new CardMatcher({ generalName: ['slash'] }).toSocketPassenger(),
-        extraUse: true,
-        commonUse: true,
-        conversation: TranslationPack.translationJsonPatcher(
-          'xuanhuo: please use slash to {0}, else {1} obtain 2 cards from you',
-          TranslationPack.patchPlayerInTranslation(room.getPlayerById(victimId)),
-          TranslationPack.patchPlayerInTranslation(room.getPlayerById(fromId)),
-        ).extract(),
-      };
+    const options: string[] = [];
+    const cards = Sanguosha.getCardsByMatcher(new CardMatcher({ generalName: ['slash'] }));
+    const attackCards: string[] = [];
 
-      const response = await room.askForCardUse(askForUseSlashTo, toId);
-      if (response.cardId !== undefined) {
-        const slashUseEvent: ServerEventFinder<GameEventIdentifiers.CardUseEvent> = {
-          fromId: response.fromId,
-          targetGroup: response.toIds && [response.toIds],
-          cardId: response.cardId,
-          triggeredBySkills: [this.Name],
-        };
-
-        await room.useCard(slashUseEvent);
-      } else {
-        await this.doSlashFail(room, fromId, toId);
+    for (const card of cards) {
+      attackCards.push(card.Name);
+    }
+    attackCards.push('duel');
+
+    for (const acard of attackCards) {
+      if (
+        room
+          .getPlayerById(first)
+          .canUseCardTo(
+            room,
+            acard !== 'duel' ? new CardMatcher({ name: [acard] }) : new CardMatcher({ generalName: [acard] }),
```

Comment:
acard is a specific card's name here, why you make it be a generalName in CardMatcher